src/components/organisms/teacher/OakUnitListOptionalityItem/OakUnitListOptionalityItem.tsx

```diff
@@ -0,0 +1,244 @@
+import React, {
+  ComponentPropsWithoutRef,
+  ElementType,
+  MutableRefObject,
+} from "react";
+
+import { OakUnitListOptionalityItemCard } from "../OakUnitListOptionalityItemCard";
+
+import {
+  OakFlex,
+  OakBox,
+  OakHeading,
+  OakSpan,
+  OakLabel,
+  OakGrid,
+  OakGridArea,
+} from "@/components/atoms";
+import { FlexStyleProps } from "@/styles/utils/flexStyle";
+
+const StyledYearAndOptionCount = ({
+  yearTitle,
+  optionalityUnitsLength,
+  disabledOrUnavailable,
+}: {
+  yearTitle: string | null | undefined;
+  optionalityUnitsLength: number;
+  disabledOrUnavailable: boolean | undefined;
+}) => (
+  <>
+    <OakSpan
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      $mr={[null, "space-between-xxxl"]}
+    >
+      {yearTitle}
+    </OakSpan>
+
+    <OakBox
+      $ph={"inner-padding-xs"}
+      $pv={"inner-padding-ssx"}
+      $ba={"border-solid-s"}
+      $borderColor={
+        disabledOrUnavailable ? "border-neutral-lighter" : "bg-decorative4-main"
+      }
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      $background={
+        disabledOrUnavailable
+          ? "bg-neutral-stronger"
+          : "bg-decorative4-very-subdued"
+      }
+      $borderRadius={"border-radius-s"}
+      $ml={[null, "space-between-s"]}
+    >
+      <OakLabel
+        $font={"heading-light-7"}
+        $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      >{`${optionalityUnitsLength} unit options`}</OakLabel>
+    </OakBox>
+  </>
+);
+
+const StyledIndex = ({
+  index,
+  disabledOrUnavailable,
+  ...rest
+}: {
+  index: number;
+  disabledOrUnavailable: boolean | undefined;
+} & FlexStyleProps) => (
+  <OakFlex
+    $btlr={"border-radius-s"}
+    $bblr={"border-radius-s"}
+    $background={disabledOrUnavailable ? "bg-neutral-stronger" : "lavender"}
+    $minWidth={"all-spacing-11"}
+    $width={["all-spacing-11", "all-spacing-11", "auto"]}
+    $height={["all-spacing-11", "auto", "auto"]}
+    $justifyContent={"center"}
+    $alignItems={"center"}
+    {...rest}
+  >
+    <OakSpan
+      $font={"heading-5"}
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+    >
+      {index}
+    </OakSpan>
+  </OakFlex>
+);
+
+const StyledUnitHeading = ({
+  nullTitle,
+  disabledOrUnavailable,
+}: {
+  nullTitle: string | undefined;
```

Comment:
🏄🏻 this is just my opinion (and it's an internal components so no biggy) but looking at LN106 `nullTitle` is the component's content so it should occupy the `children` slot.